packages/patch/src/index.ts

```diff
@@ -0,0 +1,576 @@
+import {
+  directiveUsageArgumentAdded,
+  directiveUsageArgumentDefinitionAdded,
+  directiveUsageArgumentDefinitionRemoved,
+  directiveUsageArgumentRemoved,
+  directiveUsageEnumAdded,
+  directiveUsageEnumRemoved,
+  directiveUsageEnumValueAdded,
+  directiveUsageEnumValueRemoved,
+  directiveUsageFieldAdded,
+  directiveUsageFieldDefinitionAdded,
+  directiveUsageFieldDefinitionRemoved,
+  directiveUsageFieldRemoved,
+  directiveUsageInputFieldDefinitionAdded,
+  directiveUsageInputFieldDefinitionRemoved,
+  directiveUsageInputObjectAdded,
+  directiveUsageInputObjectRemoved,
+  directiveUsageInterfaceAdded,
+  directiveUsageInterfaceRemoved,
+  directiveUsageObjectAdded,
+  directiveUsageObjectRemoved,
+  directiveUsageScalarAdded,
+  directiveUsageScalarRemoved,
+  directiveUsageSchemaAdded,
+  directiveUsageSchemaRemoved,
+  directiveUsageUnionMemberAdded,
+  directiveUsageUnionMemberRemoved,
+} from './patches/directive-usages.js';
+import {
+  directiveAdded,
+  directiveArgumentAdded,
+  directiveArgumentDefaultValueChanged,
+  directiveArgumentDescriptionChanged,
+  directiveArgumentRemoved,
+  directiveArgumentTypeChanged,
+  directiveDescriptionChanged,
+  directiveLocationAdded,
+  directiveLocationRemoved,
+  directiveRemoved,
+  directiveRepeatableAdded,
+  directiveRepeatableRemoved,
+} from './patches/directives.js';
+import {
+  enumValueAdded,
+  enumValueDeprecationReasonAdded,
+  enumValueDeprecationReasonChanged,
+  enumValueDescriptionChanged,
+  enumValueRemoved,
+} from './patches/enum.js';
+import {
+  fieldAdded,
+  fieldArgumentAdded,
+  fieldArgumentDefaultChanged,
+  fieldArgumentDescriptionChanged,
+  fieldArgumentRemoved,
+  fieldArgumentTypeChanged,
+  fieldDeprecationAdded,
+  fieldDeprecationReasonAdded,
+  fieldDeprecationReasonChanged,
+  fieldDeprecationRemoved,
+  fieldDescriptionAdded,
+  fieldDescriptionChanged,
+  fieldDescriptionRemoved,
+  fieldRemoved,
+  fieldTypeChanged,
+} from './patches/fields.js';
+import {
+  inputFieldAdded,
+  inputFieldDefaultValueChanged,
+  inputFieldDescriptionAdded,
+  inputFieldDescriptionChanged,
+  inputFieldDescriptionRemoved,
+  inputFieldRemoved,
+  inputFieldTypeChanged,
+} from './patches/inputs.js';
+import { objectTypeInterfaceAdded, objectTypeInterfaceRemoved } from './patches/interfaces.js';
+import {
+  schemaMutationTypeChanged,
+  schemaQueryTypeChanged,
+  schemaSubscriptionTypeChanged,
+} from './patches/schema.js';
+import {
+  typeAdded,
+  typeDescriptionAdded,
+  typeDescriptionChanged,
+  typeDescriptionRemoved,
+  typeRemoved,
+} from './patches/types.js';
+import { unionMemberAdded, unionMemberRemoved } from './patches/unions.js';
+import { PatchConfig, PatchContext, SchemaNode } from './types.js';
+import { debugPrintChange } from './utils.js';
+
+export * as errors from './errors.js';
+
+/**
+ * Wraps converting a schema to AST safely, patching, then rebuilding the schema from AST.
+ */
```

Comment:
Worth noting here that the schema is not validated and that it is the responsibility of the consumer of this function